Associate SIM card identifier with messages

matthewvogt opened the pull request.

amccarthy commented on modem/modem.c:
```diff
@@ -491,6 +503,29 @@ on_sim_notify_imsi (ModemSIMService *sim,
+  if (iccid == NULL)
+    return;
+
+  if (strlen (iccid))
```
iccid[0] != '\0' would be a tiny bit quicker.

monich commented:
Why ICCID rather than IMSI? IMSI is already used to associate MMS message with the SIM card, it gets stored as `"mms-notification-imsi"` event property in commhistory db. ofono stores its settings using IMSI as the key. It would seem logical to use the same key everywhere.

matthewvogt commented:
It used to be the case that getting the IMSI from the SIM was not always possible (this is from years ago...).  Can we assume that we must be able to get it from all SIMs now, or ofono would not work correctly?

monich commented:
Duplicating my email here, I would go with IMSI for consistency with MMS and ofono. I've never seen SIM card without IMSI and don't know under which conditions it may occur these days. On the other hand, the scenario with the replacement SIM having the same IMSI (and picking up existing settings if they are keyed to IMSI) is rather common. In my opinion, advantages of using IMSI overweight the potential risks.

matthewvogt committed:
[telepathy-ring] Include IMSI property in SIM messages and calls. Contributes to MER#1303

For messages and calls relating to a particular SIM, include the IMSI
value as a property named 'SubscriberIdentity'.

matthewvogt commented:
Updated to use IMSI as the identifier property.

amccarthy commented on src/ring-call-channel.c:
```diff
@@ -544,6 +553,15 @@ ring_call_channel_class_init(RingCallChannelClass *klass)
+    object_class, PROP_IMSI,
+    g_param_spec_string("imsi",
+      "IMSI",
+      "Internation Mobile Subscriber Identifer",
```
International

amccarthy replied:
Also identifier is missing an I. Wikipedia uses the work "identity", assuming that is the official definition we should probably use that.

matthewvogt replied:
This misspelling is used consistently here, I thought it might be best to preserve it.  If you're certain it's safe, we could fix all instance instead...

amccarthy replied:
Ok. The documentation is being its usual less than helpful self. Leave it as is unless someone else can shed some light.

kimmoli replied:
It is Identity here https://github.com/nemomobile/telepathy-ring/blob/9a51da9a87309542c12d6b9674b88d0369543e17/modem/sim-service.c#L282

matthewvogt replied:
True, maybe that string is just shown in a designer app or something?

matthewvogt replied:
Ok, this value is the 'blurb' here: https://developer.gnome.org/gobject/stable/gobject-GParamSpec.html#g-param-spec-internal
It is intended to be used in a tooltip or similar indicator within a property editor, so irrelevant to operation.  

I will change them all to be correct.

amccarthy commented:
Other than the name LGTM.

matthewvogt committed:
Correct the blurb for IMSI property definitions

monich commented on src/ring-call-channel.c:
```diff
@@ -633,6 +651,13 @@ ring_call_channel_fill_immutable_properties(TpBaseChannel *base,
+  if (!RING_STR_EMPTY(self->priv->imsi)) {
+    GValue* imsi = g_slice_new0 (GValue);
+    g_value_init (imsi, G_TYPE_STRING);
+    g_value_set_string (imsi, self->priv->imsi);
```
It seems that the above three lines can be replaced with `tp_g_value_slice_new_string(self->priv->imsi)`

matthewvogt replied:
Thanks, updated.

monich commented on src/ring-text-channel.c:
```diff
@@ -387,6 +402,13 @@ ring_text_channel_fill_immutable_properties (TpBaseChannel *base,
+  if (!RING_STR_EMPTY(self->priv->imsi)) {
+    GValue* imsi = g_slice_new0 (GValue);
+    g_value_init (imsi, G_TYPE_STRING);
+    g_value_set_string (imsi, self->priv->imsi);
```
Again, `tp_g_value_slice_new_string()` can save a few lines of code.

monich commented:
I find it a bit awkward that the same thing is called `"subscriber-identity"` in some places and `"imsi"` elsewhere. Generally, LGTM.

matthewvogt commented:
Yes, the property is called SubscriberIdentity in ofono so I think that is the best public name for it.  'imsi' is generally what the relevant variable was already called inside telepathy-ring, so I have continued using that internally.
